docs/proposals/33-machine-image-capabilities.md

```diff
@@ -0,0 +1,649 @@
+- The first value in the array is the most preferred value.
+- The last value is the least preferred value.
+
+Example: A machine supports hypervisor `gen2` AND `gen1` and an image version offers `gen1` OR `gen2`.
+Then the image with `gen2` will be preferred.
+
+Other edge cases will be deterministic solved as well to ensure a consistent behavior.
+
+In addition to the default capabilities, the `spec.machineTypes` is extended with the `capabilities` structure described above.
+The image versions in `spec.machineImages.versions` will be extended with `capabilityCombinations` - an array of capabilities structures, one entry for each image reference of an image version in the cloud provider.
+
+The architecture is also added to the capabilityCombinations. This is required as the architecture is a capability itself. 
+
+> [!NOTE]
+> To avoid duplication and remove the possibility of inconsistencies the existing `architecture` field in `machineImages.versions` and `machineTypes` will be marked as deprecated. `architecture` will become the only mandatory capability and will take precedence over the deprecated field if both are present.
+
+```yaml
+# CloudProfile
+spec:
+  capabilities: # <-- Full list of possible capabilities used as default
+    architecture: ["amd64", "arm64"]
+    hypervisorType: ["gen2", "gen1"]
+    network: ["accelerated", "standard"]
+
+  machineImages:
+    - name: gardenlinux
+      updateStrategy: minor
+      versions:
+        # - architectures: [arm64, amd64] # not required anymore
+        #   version: 1592.2.0-gen2
+
+        - architectures: [amd64, arm64] # <-- marked as deprecated
+          capabilitySets:
+            - architecture: [arm64] # <-- architecture must be added to the capabilities to ensure compatibility
+              hypervisorType: ["gen2"]
+              network: ["accelerated", "standard"] # <-- not required as its the default
+            - architecture: [amd64]
+              hypervisorType: ["gen2"]
+            - architecture: [amd64]
+              hypervisorType: ["gen1"]
+          classification: supported
+          version: 1592.2.0
+
+        - classification: supported
+          version: 1592.1.0 # <-- if no capabilityCombinations are defined the default capabilities are assigned as only combination
+
+  machineTypes:
+    - architecture: amd64 # <-- marked as deprecated
+      cpu: "896"
+      gpu: "0"
+      memory: 12Ti
+      name: Standard_S896om
+      usable: true
+      capabilities:
+        hypervisorType: ["gen2"] # <-- hypervisorType is overwritten
+        architecture: ["amd64"] # <-- architecture is overwritten
+
+    - cpu: "896" # <-- as no capabilities are defined the default capabilities are assigned
+      gpu: "0"
+      memory: 12Ti
+      name: Standard_S896
+      usable: true
+```
+
+The changes to the gardener `CloudProfile` above are sufficient to select the correct image version selection. This enables:
```

Comment:
This sentence reads a bit bumpy. Maybe rephrasing with `support` (or `enhance`, or `aid`) makes the intended statement clearer? Open for other suggestions.

```suggestion
The changes to the gardener `CloudProfile` above are sufficient to support the correct image version selection. This enables:
```